rio_tiler/io/base.py

```diff
@@ -92,6 +83,95 @@ def point(self, lon: float, lat: float, **kwargs: Any) -> List:
         ...
 
 
+@attr.s
+class AsyncBaseReader(metaclass=abc.ABCMeta):
```

Comment:
`abc` doesn't enforce the signature of the method, just that the method is present, so we could reuse the base class but mypy is not happy because the response type is different.

Reply:
So we just need to make sure that this base class maintains parity with the `BaseReader`?

Reply:
Yeah, there might be a better way to enforce the same type across both classes besides duck typing but I can't think of one that would make mypy happy.